src/Formatter/FormatUserMentions.php

```diff
@@ -43,10 +43,10 @@ public function __construct(SlugManager $slugManager, TranslatorInterface $trans
      */
     public function __invoke(Renderer $renderer, $context, string $xml)
     {
-        $post = $context;
-
-        return Utils::replaceAttributes($xml, 'USERMENTION', function ($attributes) use ($post) {
-            $user = $post->mentionsUsers->find($attributes['id']);
+        return Utils::replaceAttributes($xml, 'USERMENTION', function ($attributes) use ($context) {
+            $user = (isset($context->getRelations()['mentionsUsers']))
```

Comment:
thinking about it again, since `getRelations()` only returns relations that were loaded/attempted to be loaded, I think it better to also always use the relation if we're dealing with posts, so that there is no way we can fall in the performance impact by simply not having loaded the relationship. (sorry, should've thought about it the first time)

```php
            $user = (isset($context->getRelations()['mentionsUsers']) || $context instanceof Post)
```

Reply:
Is there a way to check if the relation exists on the model without checking specific classes? Thinking about perhaps other extensions implementing the relationship for their own models.

Unless we want to leave that for the proper fix.

Reply:
Looked into the inners of laravel, and searched online, doesn't seem like you can check if a model has a relationship, there's just `getRelations` which is better than nothing as it returns loaded relation names.

Reply:
No problem, updated!